src/gui/editorwidgets/qgsrelationreferencewidgetwrapper.cpp

```diff
@@ -71,7 +71,7 @@ void QgsRelationReferenceWidgetWrapper::initWidget( QWidget* editor )
     if ( ctx->relation().name() == relation.name() )
     {
       mWidget->setEmbedForm( false );
-      mWidget->setReadOnlySelector( false );
+      mWidget->setReadOnlySelector( true );
```

Comment:
@m-kuhn 

The comment on this part says `// If this widget is already embedded by the same relation, reduce functionality`. So it seems that the widget should be read only to correctly reduce its functionality. Can you give me your opinion?

Reply:
Sorry, can you repeat the question, I couldn't follow, sorry.

Reply:
@pblottiere the `readOnly` only applies to the selector (i.e. a line edit instead of a combobox) but not the whole widget.

Reply:
Sorry @m-kuhn , it's my fault.

Actually I was searching why a `QgsRelationReferenceWidget` in a referenced layer (parent layer in the relation) form do not use the read only configuration currently defined and it seems that it's due to this part of the code:

``` c++
  // If this widget is already embedded by the same relation, reduce functionality
  const QgsAttributeEditorContext* ctx = &context();
  do
  {
    if ( ctx->relation().name() == relation.name() )
    {
      mWidget->setEmbedForm( false );
      mWidget->setReadOnlySelector( false );
      mWidget->setAllowMapIdentification( false );
    }
    ctx = ctx->parentContext();
}
```

From what I understand, if we want to reduce functionality of the widget as the comment is implying, then the widget should be in `readOnly` mode.

@3nids 

> he readOnly only applies to the selector (i.e. a line edit instead of a combobox) but not the whole widget.

True. But when a relation reference widget is not in read only mode, the combobox is populated with all possible values. But doing that is very (very) time consuming  when the parent layer has a lot of features.

So I was just wondering if the widget should be in `readOnly` mode in this case...

Let me know what you think :)

Reply:
Sounds reasonable. I guess there's a noticeable speed improvement?

Reply:
Indeed... Several tens of seconds with the (big) database of @haubourg!

Reply:
go for it! i am even considering to hide it alrogether